### PR TITLE
executor: Count the number of hash table collisions and build elapse during Hash Join.

### LINKED ISSUE #14387: Count the number of hash table collisions and failures during Hash Join.
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently we use native [FNV](https://golang.org/pkg/hash/fnv/#New64) to build hash table. We need measure whether the hash function is effective with real dataset.

**Describe the feature you'd like:**

I propose to add three metrics into `explain analyze` result in each hash join.
1. Count of hash collisions when probing, which means we do extra scans for hashed values.
2. Count of hash failures when probing, which means we do unused scans for hashed values and keys when hash key is not matched.
3. Time cost to finish building hash table in a single goroutine.

After that, we can evaluate whether the hash function we currently used is suitable and can do experiments for other hash functions.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

Close #14387

### What is changed and how it works?

We can see `collision:0, build:3.418µs`

```
mysql> explain analyze select t1.b from t t1, t t2 where t1.a = t2.a;
+----------------------------+-------+-----------+------------------------------------------------------------------+--------------------------------------------------------------------------------+---------------+---------+
| id                         | count | task      | operator info                                                    | execution info                                                                 | memory        | disk    |
+----------------------------+-------+-----------+------------------------------------------------------------------+--------------------------------------------------------------------------------+---------------+---------+
| Projection_7               | 1.00  | root      | test.t.b                                                         | time:663.604µs, loops:2, rows:1, Concurrency:OFF                               | 1.08984375 KB | N/A     |
| └─HashLeftJoin_8           | 1.00  | root      | inner join, inner:TableReader_15, equal:[eq(test.t.a, test.t.a)] | time:660.333µs, loops:2, rows:1, Concurrency:5, collision:0, build:3.418µs     | 8.234375 KB   | 0 Bytes |
|   ├─TableReader_12         | 1.00  | root      | data:Selection_11                                                | time:577.456µs, loops:2, rows:1, rpc num: 1, rpc time:547.891µs, proc keys:1   | 195 Bytes     | N/A     |
|   │ └─Selection_11         | 1.00  | cop[tikv] | not(isnull(test.t.a))                                            | time:0s, loops:1, rows:1                                                       | N/A           | N/A     |
|   │   └─TableScan_10       | 1.00  | cop[tikv] | table:t1, range:[-inf,+inf], keep order:false, stats:pseudo      | time:0s, loops:1, rows:1                                                       | N/A           | N/A     |
|   └─TableReader_15         | 1.00  | root      | data:Selection_14                                                | time:536.444µs, loops:2, rows:1, rpc num: 1, rpc time:502.082µs, proc keys:1   | 179 Bytes     | N/A     |
|     └─Selection_14         | 1.00  | cop[tikv] | not(isnull(test.t.a))                                            | time:0s, loops:1, rows:1                                                       | N/A           | N/A     |
|       └─TableScan_13       | 1.00  | cop[tikv] | table:t2, range:[-inf,+inf], keep order:false, stats:pseudo      | time:0s, loops:1, rows:1                                                       | N/A           | N/A     |
+----------------------------+-------+-----------+------------------------------------------------------------------+--------------------------------------------------------------------------------+---------------+---------+
8 rows in set (0.00 sec)
```

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/hash_table.go [hunk 1]
```diff
@@ -130,6 +142,7 @@ func (c *hashRowContainer) GetMatchedRowsAndPtrs(probeKey uint64, probeRow chunk
 			return
 		}
 		if !ok {
+			c.stat.collision++
```
Comment (zz-jason) [inline executor/hash_table.go]: this `collision` does not mean the hash collision in the build tale. users may get confused about this message.

Comment (SunRunAway) [reply executor/hash_table.go]: This is exactly the hash collision in the build table.

Comment (zz-jason) [reply executor/hash_table.go]: please change the output info to `probe collision`

**Pull request merged**